src/tests/tests/asan-error-parse.rkt

```diff
@@ -0,0 +1,248 @@
+      (define student-code #<<HERE
+void h(int *p) { p[150] = 99999; free(p); }
+int main() {
+    h(malloc(100));
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "heap-buffer-overflow"))
+
+;; ---- DOUBLE FREE TEST ---------------------------
+    (test-case "Double Free Test"
+      (define student-code #<<HERE
+void *get100bytes() {
+    void *p = malloc(100);
+    free(p);
+    return p;
+}
+int main() {
+    int *p = get100bytes();
+    free(p);
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "double-free")
+      (check-equal? (length (hash-ref json-answer 'call_stacks))  3))
+
+;; ---- HEAP USE AFTER FREE TEST ---------------------------
+    (test-case "Heap Use After Free Test"
+      (define student-code #<<HERE
+void *get100bytes() { return malloc(100); }
+int main() {
+    int *p = get100bytes();
+    free(p);
+    p[10] = 999;
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "heap-use-after-free")
+      (check-equal? (length (hash-ref json-answer 'call_stacks))  3))
+
+
+;; ---- GLOBAL MEMORY TEST ---------------------------
+    (test-case "Global Memory Test"
+      (define student-code #<<HERE
+int my_array[100];
+int main() {
+    my_array[120] = 9999;
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "global-buffer-overflow"))
+
+;; ---- SEGMENTATION FAULT TESTS ---------------------------
+    (test-case "Segmentation Fault Null Test"
+      (define student-code #<<HERE
+void f(char *x) { *x = 'a'; }
+int main() {
+    f(0);
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "segmentation-fault-on-null-address"))
+
+    (test-case "Segmentation Fault Test"
+      (define student-code #<<HERE
+int main() {
+    int *p = (int*) main;
+    while(1) { *p++ = 0xdeadbeef; }
+    return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "segmentation-fault"))
+
+;; ---- NO ERROR TEST ---------------------------
+    (test-case "No Error Test"
+      (define student-code #<<HERE
+#include <stdlib.h>
+
+int global[10];
+
+void f(char *a, int len) {
+  for(int i = 0; i < len; i++) {
+    a[i] = 'x';
+  }
+}
+
+int main() {
+  double *p = malloc(100 * sizeof(p));
+  for(int i = 0; i < 100; i++) {
+    p[i] = 0.123 * i;
+  }
+  free(p);
+  p = NULL;
+  return 0;
+}
+HERE
+)
+      (define json-answer (compile-run-wait student-code))
+      (check-equal? (hash-ref json-answer 'error_type) "unknown")
+      (check-equal? (hash-ref json-answer 'raw_message) ""))
+
+))
```

Comment:
Looks good, please add additional test for comment above on stack variable information.